Slim/Control/Queries.pm

```diff
@@ -4734,24 +4739,48 @@ sub worksQuery {
 		$sql .= ' ';
 	}
 
-	$sql .= " GROUP BY $groupBy ";
-
 	my $dbh = Slim::Schema->dbh;
 
+	my $stillScanning = Slim::Music::Import->stillScanning();
+
 	# Get count of unique composers, the count is cached until the next rescan done event
-	my $cacheKey = md5_hex($sql . join( '', @{$p} ) . 'composerCount' . Slim::Music::VirtualLibraries->getLibraryIdForClient($client));
+	my $ccSql = sprintf($sql . " GROUP BY composer.id", "'c'");
+	my $cacheKey = md5_hex($ccSql . join( '', @{$p} ) . Slim::Music::VirtualLibraries->getLibraryIdForClient($client));
 
 	my $composerCount = $cache->{$cacheKey};
 	if ( !$composerCount ) {
-		my $ccSql = sprintf($sql, "composer.id AS cid");
 		my $total_sth = $dbh->prepare_cached( qq{
-			SELECT COUNT(DISTINCT cid)  FROM ( $ccSql ) AS t1
+			SELECT COUNT(1)  FROM ( $ccSql ) AS t1
 		} );
 
 		$total_sth->execute( @{$p} );
 		($composerCount) = $total_sth->fetchrow_array();
 		$total_sth->finish;
 	}
+	if ( !$stillScanning ) {
+		$cache->{$cacheKey} = $composerCount;
+	}
+
+
+	$sql .= " GROUP BY tracks.work, composer.id ";
+
+	# Get count of all results, the count is cached until the next rescan done event
+	my $cSql = sprintf($sql, "'c'");
+	$cacheKey = md5_hex($cSql . join( '', @{$p} ) . Slim::Music::VirtualLibraries->getLibraryIdForClient($client));
+
+	my $count = $cache->{$cacheKey};
+	if ( !$count ) {
+		my $total_sth = $dbh->prepare_cached( qq{
+			SELECT COUNT(1) FROM ( $cSql ) AS t1
+		} );
+
+		$total_sth->execute( @{$p} );
+		($count) = $total_sth->fetchrow_array();
+		$total_sth->finish;
+	}
+	if ( !$stillScanning ) {
+		$cache->{$cacheKey} = $count;
+	}
```

Comment:
same here - thanks!

Reply:
OK, will do. Do you want me to make the same change in albumsQuery, etc?

Reply:
Oh my... never mind. I'll merge and do them all. Thanks for the heads up (once again)!